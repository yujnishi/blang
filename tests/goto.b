main() {
    auto l 0;

    printf("start: %d*n",l);

label1:
    printf("label1: %d*n",l);
    if ( l++ == 0 ) goto label1;

label2:
    printf("label2: %d*n",l);
    if ( l++ == 1 ) goto label1;
    if ( l++ == 3 ) goto label2;

label3:
    printf("label3: %d*n",l);
    if ( l++ <= 9 ) goto label3;

    printf("end: %d*n",l);
}
