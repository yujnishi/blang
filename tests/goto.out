start: 0
label1: 0
label1: 1
label2: 2
label2: 4
label3: 6
label3: 7
label3: 8
label3: 9
label3: 10
end: 11
